fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/dao/ReindexResourceDAO.java

```diff
@@ -267,6 +319,17 @@ protected ResourceIndexRecord getNextResource(SecureRandom random, Instant reind
             }
         } while (offsetRange > 0 && result == null);
 
+        if (result != null) {
+            // Since V0027 we lock resources using LOGICAL_RESOURCE_IDENT. There is a small chance of
+            // deadlock here if reindex is processing the same resources which are also being updated
+            // as part of ingestion. This is because ingestion locks LOGICAL_RESOURCE_IDENT before
+            // updating LOGICAL_RESOURCES, whereas here we updated LOGICAL_RESOURCES before locking
+            // LOGICAL_RESOURCE_IDENT. Eliminating this would require a significant rework. The
+            // recommendation going forward is to always use the client-driven process, because it
+            // is more efficient.
```

Comment:
I wonder if we should deprecate (or remove) the server-driven one...

Personally I do like the simplicity of being able to just call `[base]/[resourceType]/$reindex`.
In fact I was sort of wondering if we could support `[base]/[resourceType]/$reindex?timestamp=[isoDateTime]` to make it even easier... (today we don't support that on POST requests, only GET ones...and we force reindex to use POST)